frame/sgx-hello-world/src/lib.rs

```diff
@@ -383,39 +388,67 @@ impl<T: Trait> Module<T> {
 		Ok(())
 	}
 
-	fn dispatch_waiting_calls(block_number: T::BlockNumber, signer: &Signer<T, T::AuthorityId, frame_system::offchain::ForAny>) -> Result<(), &'static str> {
+	fn dispatch_waiting_calls(
+		block_number: T::BlockNumber,
+		signer: &Signer<T, T::AuthorityId, frame_system::offchain::ForAny>
+	) -> Result<(), &'static str> {
 		debug::trace!(target: "sgx", "[dispatch_waiting_calls] START at block_number: {:?}", block_number);
-
+		let pending_calls = <WaitingEnclaveCalls<T>>::get();
+		debug::trace!(target: "sgx", "[dispatch_waiting_calls, #{:?}] Pending enclave calls: {}", block_number, pending_calls.len());
 		let mut dispatched = Vec::new();
+		let mut fail_count = 0;
 
-		for (enclave_id, xt) in <WaitingEnclaveCalls<T>>::get() {
+		for (enclave_id, xt) in pending_calls {
```

Comment:
I think we might want to refactor this to execute calls one by one and return at the first error?

Reply:
I'm not convinced by that, why should the potential other calls not get the chance because one failed? Or are saying that the enclave is likely unresponsive or down?

Reply:
What I was getting at is that it may be better code to execute each call in its own function rather than in a loop like this, especially for error handling.
You are right though that there are probably some errors, like `500 INTERNAL SERVER ERROR`, that we could consider fatal and be like "hey, that enclave isn't doing well, let's retry in a 100 blocks".